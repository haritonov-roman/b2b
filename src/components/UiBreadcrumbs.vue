<script>
export default {
  name: 'UiBreadcrumbs',

  props: {
    routes: {
      type: Array,
      required: true,
      default: () => []
    }
  }
}
</script>

<template>
  <cds-breadcrumbs
    :items="routes"
    class="breadcrumbs"
  />
</template>

<style lang="scss" scoped>
.breadcrumbs {
  margin-bottom: 1rem;
}
</style>
