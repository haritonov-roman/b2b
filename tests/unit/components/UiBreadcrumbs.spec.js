import { createLocalVue, shallowMount } from '@vue/test-utils'
import Cds from '@central-design-system/components'
import UiBreadcrumbs from '@/components/UiBreadcrumbs'

const localVue = createLocalVue()

localVue.use(Cds)

describe('UiBreadcrumbs.vue', () => {
  const routes = []

  const wrapper = shallowMount(UiBreadcrumbs, {
    propsData: { routes },
    localVue
  })

  it('Проверка рендеринга', () => {
    expect(wrapper.exists()).toBe(true)
  })

  it('Проверка атрибута items', () => {
    expect(wrapper.attributes('items')).toMatch('')
  })

  it('Снимок', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
