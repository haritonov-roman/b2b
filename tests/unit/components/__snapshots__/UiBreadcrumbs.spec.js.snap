// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UiBreadcrumbs.vue Снимок 1`] = `
<cds-breadcrumbs-stub
  class="breadcrumbs"
  items=""
/>
`;
